### ISSUE TITLE
tests fail on linux/x86_64 with go 1.9

### ISSUE THREAD

**Issue by decathorpe** [sasha-s/go-deadlock#3]
When running "go test" with go 1.9 (linux/x86_64, other architectures seem to work), the following failure occurs:

```
+ go test -compiler gc -ldflags '' github.com/sasha-s/go-deadlock
POTENTIAL DEADLOCK: Inconsistent locking. saw this ordering in one goroutine:
happened before
deadlock_test.go:44 go-deadlock.TestNoDeadlocks.func2.1 { a.RLock() } <<<<<
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
happened after
deadlock_test.go:47 go-deadlock.TestNoDeadlocks.func2.1.1 { b.Lock() } <<<<<
deadlock_test.go:54 go-deadlock.TestNoDeadlocks.func2.1 { }() }
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
in another goroutine: happened before
deadlock_test.go:47 go-deadlock.TestNoDeadlocks.func2.1.1 { b.Lock() } <<<<<
deadlock_test.go:54 go-deadlock.TestNoDeadlocks.func2.1 { }() }
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
happend after
deadlock_test.go:44 go-deadlock.TestNoDeadlocks.func2.1 { a.RLock() } <<<<<
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
Other goroutines holding locks:
goroutine 0 lock 0xc420014118
deadlock_test.go:47 go-deadlock.TestNoDeadlocks.func2.1.1 { b.Lock() } <<<<<
deadlock_test.go:54 go-deadlock.TestNoDeadlocks.func2.1 { }() }
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
goroutine 0 lock 0xc4200160e0
deadlock_test.go:50 go-deadlock.TestNoDeadlocks.func2.1.1.1 { c.Lock() } <<<<<
deadlock_test.go:53 go-deadlock.TestNoDeadlocks.func2.1.1 { }() }
deadlock_test.go:54 go-deadlock.TestNoDeadlocks.func2.1 { }() }
deadlock_test.go:55 go-deadlock.TestNoDeadlocks.func2 { }() }
FAIL	github.com/sasha-s/go-deadlock	0.003s
```

**Comment by sasha-s** [comment]
Could not reproduce it with `go1.9rc1 linux/amd64`, Xenial/docker/Mac.
```
....
2017/08/05 19:15:02 ****************
PASS
ok  	github.com/sasha-s/go-deadlock	0.987s
```
What versions are you using?

**Comment by sasha-s** [comment]
Added 1.9rc1 to travis-ci. Works. https://travis-ci.org/sasha-s/go-deadlock/jobs/261375190

**Comment by decathorpe** [comment]
That's strange, because both fedora builds of the package failed on x86_64 (and armhfp) since go was updated to version 1.9beta2.

If it helps, full build logs for the builds with failed tests are available at:

- https://koji.fedoraproject.org/koji/taskinfo?taskID=20758842
- https://koji.fedoraproject.org/koji/taskinfo?taskID=20989298

**Comment by sasha-s** [comment]
It seems the goroutine id detection https://github.com/petermattis/goid is broken in 1.9beta2.
I am inclined to wait till 1.9 lands -- maybe it would just work out of the box.
Or I'll fix it.

**Comment by decathorpe** [comment]
I just ran the tests again atop the latest git master of petemattis/goid (which should fix problems with go 1.9), and it works now ... Thanks for pointing me in the right direction, and sorry for the "Issue noise" :+1:
